### DISCUSSION TITLE
the actual roles of different binds

### DISCUSSION THREAD

**Discussion by zhoujingfighting** [eclipse-theia/theia#9601] [Q&A] [answered]
I get confused about the bind
in some case , 
```js
 bind(ElectronMenuUpdater).toSelf().inSingletonScope();
```
in some case ,
```js
 bind(SampleUpdaterClient).toService(SampleUpdaterClientImpl);
```
in other case ,
```js
 bind(SampleUpdater).toDynamicValue(context => {
        const client = context.container.get(SampleUpdaterClientImpl);
        return ElectronIpcConnectionProvider.createProxy(context.container, SampleUpdaterPath, client);
    }).inSingletonScope();
```
so , any links about explaining different bindings??

**Comment by vince-fugnitto** [comment, marked as answer]
@zhoujingfighting the framework uses [Inversify JS](https://inversify.io/) for dependency injection, you can take a look at their documentation to understand the different types of bindings further:
- https://github.com/inversify/InversifyJS/tree/master/wiki
